src/photos/ducks/clustering/settings.js

```diff
@@ -3,33 +3,64 @@ import { DOCTYPE_PHOTOS_SETTINGS } from 'drive/lib/doctypes'
 import {
   SETTING_TYPE,
   DEFAULT_SETTING,
-  DEFAULT_PERIOD,
-  DEFAULT_MODE
+  DEFAULT_EPS_TEMPORAL,
+  DEFAULT_EPS_SPATIAL,
+  DEFAULT_MAX_BOUND,
+  DEFAULT_MODE,
+  COARSE_COEFFICIENT
 } from './consts'
+import { gradientAngle } from 'photos/ducks/clustering/gradient'
+import { matchingParameters } from './matching'
 
-export const createDefaultSetting = () => {
-  log('info', 'Create default setting')
-  return cozyClient.data.create(DOCTYPE_PHOTOS_SETTINGS, DEFAULT_SETTING)
+export const createSetting = initParameters => {
+  log('info', 'Create setting')
+  const defaultSetting = DEFAULT_SETTING
+  defaultSetting.parameters[0] = initParameters
+  return cozyClient.data.create(DOCTYPE_PHOTOS_SETTINGS, defaultSetting)
 }
 
 export const readSetting = async () => {
   const settings = await cozyClient.data.findAll(DOCTYPE_PHOTOS_SETTINGS)
   return settings.find(doc => doc.type === SETTING_TYPE)
 }
 
-export const defaultParameters = setting => {
-  const param = setting.parameters.filter(param => {
-    return param.period === DEFAULT_PERIOD
-  })
-  const defaultParam =
-    param.length > 0
-      ? param[0].modes.find(mode => mode.name === DEFAULT_MODE)
-      : null
-  return defaultParam
-    ? {
-        eps: defaultParam.eps,
-        epsTemporal: defaultParam.eps_temporal,
-        epsSpatial: defaultParam.eps_spatial
+const getDefaultParameters = params => {
+  const mode = params.modes.find(mode => mode.name === DEFAULT_MODE)
+  if (!mode || !mode.epsTemporal || !mode.epsSpatial) {
+    return null
+  }
+  mode.epsMax = Math.max(mode.epsTemporal, mode.epsSpatial)
+  mode.maxBound =
+    mode.epsMax * 2 < DEFAULT_MAX_BOUND ? mode.epsMax * 2 : DEFAULT_MAX_BOUND
+  mode.cosAngle = gradientAngle(mode.epsMax, COARSE_COEFFICIENT)
```

Comment:
Is it safe to directly modify an object inside `params.modes`?

Reply:
We wouldn't be? We need the params written in the db and computes some other ones based on them.

Reply:
The risk is that another function somewhere in the flow will use the same object, and behaves differently because we modified it here. I don't think it's the case here, but it's generally something to be mindful of.